README.md

```diff
@@ -306,6 +307,50 @@ With precompiled bundle:
 </script>
 ```
 
+## Binary data
+QR Codes can hold arbitrary byte-based binary data. If you attempt to create a binary QR Code by first converting the data to a JavaScript string, it will fail to encode propery because string encoding adds additional bytes. Instead, you must pass a [`Uint8ClampedArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8ClampedArray) or compatible array, or a Node [Buffer](https://nodejs.org/api/buffer.html), as follows:
+
+```javascript
+// Regular array example
+// WARNING: Element values will be clamped to 0-255 even if your data contains higher values.
+const QRCode = require('qrcode')
+QRCode.toFile(
+  'foo.png',
+  [{ data: [253,254,255], mode: 'byte' }],
+  ...options...,
+  ...callback...
+)
+```
+
+```javascript
+// Uint8ClampedArray example
+const QRCode = require('qrcode')
+
+QRCode.toFile(
+  'foo.png',
+  [{ data: new Uint8ClampedArray([253,254,255]), mode: 'byte' }],
+  ...options...,
+  ...callback...
+)
+```
+
+```javascript
+// Node Buffer example
+// WARNING: Element values will be clamped to 0-255 even if your data contains higher values.
+const QRCode = require('qrcode')
+
+QRCode.toFile(
+  'foo.png',
+  [{ data: Buffer.from([253,254,255]), mode: 'byte' }],
+  ...options...,
+  ...callback...
+)
+```
+
+Note: binary encoding is only available on the server due to this library using Node `Buffer` internally.
```

Comment:
i think we polyfill it for the bundle so it likely works on the client too. but its clearly more useful when you have a node buffer laying about.